OpenRA.Game/Graphics/CursorManager.cs

```diff
@@ -168,8 +169,14 @@ public void Tick()
 
 		void Update()
 		{
-			if (cursor != null && frame >= cursor.Cursors.Length)
-				frame %= cursor.Cursors.Length;
+			if (cursor != null)
+			{
+				var length = cursor.Cursors.Length;
+				if (length == 0)
+					throw new InvalidDataException($"Invalid cursor definition: {cursor.Name}");
```

Comment:
Can we move this to the constructor instead? We can get the length of frames already here:
https://github.com/OpenRA/OpenRA/blob/50b529145a44d94428281d8c0fb262e4fb0569ac/OpenRA.Game/Graphics/CursorManager.cs#L50

Reply:
No, that just creates
```
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object.
   at OpenRA.Graphics.CursorManager..ctor(CursorProvider cursorProvider) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Graphics/CursorManager.cs:line 32
   at OpenRA.Game.InitializeMod(String mod, Arguments args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 492
   at OpenRA.Game.Initialize(Arguments args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 438
   at OpenRA.Game.InitializeAndRun(String[] args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Game/Game.cs:line 307
   at OpenRA.Launcher.Program.Main(String[] args) in /home/matthias/Entwicklung/OpenRA/OpenRA/OpenRA.Launcher/Program.cs:line 32
```

which is not helpful at all.

Reply:
We literally set `Cursors = new IHardwareCursor[frames.Length]` in the ctor, so I don't understand why we can't do what we do here in the constructor instead.

Reply:
Wait, this actually does work and it crashes at startup instead of cursor usage which I think is beneficiary.